microprot/scripts/pdb_search.py

```diff
@@ -388,3 +391,109 @@ def report_uncovered_subsequences(hits, query, min_subseq_len=40):
 
     return [u for u in uncovered if (u['sequence'] != '') &
                                     (len(u['sequence']) >= min_subseq_len)]
+
+
+def mask_sequence(hhsuite_fp, fullsequence_fp, subsequences_fp=None,
+                  min_prob=None, max_pvalue=None, max_evalue=None,
+                  min_fragment_length=None):
+    """ Splits a protein sequence according to HHsuits results.
+
+    The returned sub-sequences will seamlessly build the full sequence if
+    re-concatenated.
+
+    Parameters
+    ----------
+    hhsuite_fp : str
+        Filepath to HHblits/HHsearch output.
+    fullsequence_fp : str
+        Filepath to the protein sequence of the original query.
+    subsequences_fp : str
+        Filepath to which sub-sequences are written as a multiple fasta file.
+        Each sequence makes up one header and one sequence file, i.e. sequences
+        are not wrapped.
+        Default: None, i.e. no file is written.
+    min_prob: float
+        Minimal probability of a hit to be included in the resulting list.
+        Note: probabilities are in the range of 100.0 to 0.0.
+        Default: None, i.e. no filtering on probability.
+    max_pvalue: float
+        Maximal P-value of a hit to be included in the resulting list.
+        Default: None, i.e. no filtering on P-value.
+    max_evalue: float
+        Maximal E-value of a hit to be included in the resulting list.
+        Default: None, i.e. no filtering on E-value.
+    min_fragment_length: int
+        Minimal fragment length of a hit to be included in the resulting list.
+        Default: None, i.e. no filtering on fragment length.
+
+    Returns
+    -------
+    [(str, str)] where first component is a fasta header, while the second is
+    its fasta sequence.
+
+    Raises
+    ------
+    IOError
+        If the file cannot be written.
+
+    Notes
+    -----
+    A hit must satisfy ALL filtering options (min_prob, max_pvalue, max_evalue,
+    min_fragment_length) to be included in the resulting list.
+    """
+
+    # parse hits from file
+    hits = parse_pdb_match(hhsuite_fp)
+
+    # filter hits
+    if min_prob is not None:
+        hits = [hit for hit in hits if hit['Probab'] >= min_prob]
+    if max_pvalue is not None:
+        hits = [hit for hit in hits if hit['P-value'] <= max_pvalue]
+    if max_evalue is not None:
+        hits = [hit for hit in hits if hit['E-value'] <= max_evalue]
+    if min_fragment_length is not None:
+        def frag_size(hit):
+            return (hit['alignment']['Q Consensus']['end'] -
+                    hit['alignment']['Q Consensus']['start'] +
```

Comment:
Here it should compare the *actual* sequence, not the consensus, e.g.
```
hit['alignment']['Q gi|556503834|r']['end'] -
hit['alignment']['Q gi|556503834|r']['start'] +
```
I realize this might be a pain in the ass, because the name changes based on the input, but there is no guarantee that those 2 are equivalent for distant homologs.